lib/MusicBrainz/Server/View/Node.pm

```diff
@@ -8,12 +8,18 @@ use Encode;
 use HTML::Entities qw( decode_entities );
 use HTTP::Request;
 use JSON -convert_blessed_universally;
-use MusicBrainz::Server::Data::Utils qw( boolean_to_json );
+use MusicBrainz::Server::Data::Utils qw( boolean_to_json generate_token );
 use URI;
+use URI::QueryParam;
+
+use feature 'state';
 
 sub process {
     my ($self, $c) = @_;
 
+    state $server_token = generate_token();
+    state $request_id = 0;
```

Comment:
Should be safe. It definitely gets incremented & persists across requests, if that's what you mean (made sure to test that). & Each process will have it's own state variable here, and a process can only be handling one request at a time.